rust/src/core.rs

```diff
@@ -163,6 +166,83 @@ impl FilesystemScriptPrep {
     }
 }
 
+/// Some Fedora/RHEL kernels ship .hmac files with absolute paths inside,
+/// which breaks when we relocate them into ostree/.  This function
+/// changes them to be relative.
+///
+/// Fixes are pending to ensure the kernels are doing this from the start,
+/// but we can't rely on that landing for a while. In addition, there are
+/// many 3rd-party kernels that have replicated the behavior in the ARK
+/// kernel.spec.
+fn verify_kernel_hmac_impl(moddir: &Dir) -> Result<()> {
+    use std::os::unix::ffi::OsStrExt;
+
+    // FIXME: in 2023
+    // This method is intentionally a misnomer because it should eventually
+    // be changed to the "sanity check" (below). It currently patches absolute
+    // paths to give kernel package maintainers time to update their .spec files.
+
+    const SEPARATOR: &[u8] = b"  ";
+
+    let hmac_path = ".vmlinuz.hmac";
+
+    let hmac_contents = if let Some(f) = moddir.open_optional(hmac_path)? {
+        let mut f = BufReader::new(f);
+        let mut s = Vec::new();
+        f.read_to_end(&mut s)?;
+        s
+    } else {
+        return Ok(());
+    };
+
+    if !hmac_contents.contains(&b'/') {
+        return Ok(());
+    }
+
+    let split_index = hmac_contents
+        .windows(SEPARATOR.len())
+        .position(|v| v == SEPARATOR)
+        .ok_or_else(|| {
+            anyhow::anyhow!(
+                "Missing path in .vmlinuz.hmac: {}",
+                String::from_utf8_lossy(hmac_contents.as_slice())
```

Comment:
Then if `hmac_contents` was a `String` from the start we wouldn't need to duplicate this bits in the error handling path.  (And actually with Rust 1.58 we can just say `anyhow::anyhow!("Missing path in .vmlinuz.hmac: {hmac_contents}")`)

Reply:
I did try the Rust 1.58 syntax but the compiler complained about it. I think anyhow might need `format_args!` (or an upgrade).